api_app/api/routes/costs.py

```diff
@@ -15,7 +16,8 @@
 from models.domain.costs import CostReport, GranularityEnum, WorkspaceCostReport
 from resources import strings
 from services.authentication import get_current_admin_user, get_current_workspace_owner_or_tre_admin
-from services.cost_service import CostService, SubscriptionNotSupported, WorkspaceDoesNotExist
+from services.cost_service import CostService, ServiceUnavailable, SubscriptionNotSupported, TooManyRequests, WorkspaceDoesNotExist
+from starlette.responses import JSONResponse
```

Comment:
Why are you importing from starlette?
Generally, FastAPI has it's own types (based upon the starlette ones) that you should use, rather than importing the underlying starlette type. In this case, the one in fastapi.responses.

Reply:
you are correct I didn't notice that, I saw we use those types elsewhere in the code inside: ```/api_app/api/errors/generic_error.py``` ```starlette.responses.PlainTextResponse```

I'll replace this type to FastAPI's types.

Reply:
Yeah....these types slip in to our code from time to time, but it's generally not needed. FastAPI exposes types that wrap these things, and it's safer (for us) to try and stick to using the FastAPI types.